src/particle/particle/particle.py

```diff
@@ -1226,38 +1225,6 @@ def findall(
             cls.finditer(filter_fn=filter_fn, particle=particle, **search_terms)
         )
 
-    @classmethod
```

Comment:
While it is good to remove the legacy method, I also like to make it an explicit error in a first round (just a RuntimeError which says it was removed).

The motivation for this is that users tend to ignore deprecation warnings in my experience (developers don't). So the transition doesn't happen when it's deprecated, but when it fails. And an error saying it was removed is more user-friendly than a "method not found"

(I see a function raising a RuntimeError as "removed", physically removing it can still happen any time, it's already not "there" anymore)

but It's a bit of a style, however you prefer. I can imagine a lot of folks using this function somewhere (with non-updated code) and being surprised when it fails without a clear instruction on how to update.

Reply:
Oh, I see. I did not think about it. Have no strong opinion. What you suggest certainly makes very much sense for a package with a large user base. Here I dunno.

You mean something like
```
@classmethod
    def find(cls, *args, **search_terms):
        # type: (Any, Any) -> Particle
        raise RuntimeError("Method has been removed post versions 0.16. Please use finditer or findall instead.")
```
?

Reply:
You are reminding me that we need to move type comments into annotations. :)

That would work, no strong opinion, but I do see where it might be slightly nicer, prompting users in the right direction. I'd use `AttributeError`, though, so it doesn't modify behavior.

Reply:
OK, will make this kind of update tomorrow ...

Reply:
Yep exactly, and indeed, the AttributeError could make sense. As said, not cruciol, however you feel like

Reply:
Added now.